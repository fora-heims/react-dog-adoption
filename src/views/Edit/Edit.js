import './Edit.css';

import { useState, useEffect, React } from 'react';
import Header from '../../components/Header/Header.js';
import Form from '../../components/Form/Form.js';
import { getDog } from '../../services/dogs.js';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';

export default function Edit() {
  const [dogInfo, setDogInfo] = useState({});

  let params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDog(params.id);
      setDogInfo(data);
    };
    fetchData();
  }, [params.id]);

  const saveButtonHandler = () => {};

  return (
    <div>
      <Header />
      <Form handler={saveButtonHandler} dog={dogInfo} setDogInfo={setDogInfo} />
    </div>
  );
}
