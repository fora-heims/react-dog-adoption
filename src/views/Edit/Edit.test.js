import Edit from './Edit.js';
import { MemoryRouter, Route } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

test('', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/1/edit']}>
      <Route path="/dogs/:id/edit">
        <Edit />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText(
    "Please don't mess with Dragon. It might break this site (snapshot testing)."
  );
  expect(container).toMatchSnapshot();
});
