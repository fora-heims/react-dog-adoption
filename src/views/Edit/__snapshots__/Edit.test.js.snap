// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div
    class="edit"
  >
    <header>
      <a
        class="navigate activenot-active navigate"
        href="/"
      >
        Home
      </a>
      <a
        class="navigate activenot-active navigate"
        href="/admin"
      >
        Admin
      </a>
    </header>
    <div
      class="form"
    >
      <h2>
        Enter Dog Details
      </h2>
      <p />
      <label>
        Name:
        <input
          placeholder="Name"
          type="text"
          value="Dragon"
        />
      </label>
      <label>
        Bio:
        <textarea
          placeholder="Bio"
        >
          Please don't mess with Dragon. It might break this site (snapshot testing).
        </textarea>
      </label>
      <label>
        Age:
        <input
          placeholder="Age"
          type="number"
          value="267"
        />
      </label>
      <label>
        Image:
        <input
          placeholder="Image URL"
          type="text"
          value="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Gringotts_Dragon_%2842428011465%29.jpg/675px-Gringotts_Dragon_%2842428011465%29.jpg"
        />
      </label>
      <label>
        Breed:
        <input
          placeholder="Breed"
          type="text"
          value="Ukrainian Ironbelly"
        />
      </label>
      <button
        class="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
